packages/less/test/less-test.js

```diff
@@ -169,6 +169,29 @@ module.exports = function() {
         }
     }
 
+    function testSourcemapWithVariableInSelector(name, err, compiledLess, doReplacements, sourcemap, baseFolder) {
+        if (err) {
+            fail('ERROR: ' + (err && err.message));
+            return;
+        }
+
+        // Even if annotation is not necessary, the map file should be there.
+        fs.readFile(path.join('test/', name) + '.json', 'utf8', function (e, expectedSourcemap) {
+            process.stdout.write('- ' + path.join(baseFolder, name) + ': ');
+            if (sourcemap === expectedSourcemap) {
+                ok('OK');
+            } else if (err) {
+                fail('ERROR: ' + (err && err.message));
```

Comment:
I copied this from another source map test: https://github.com/less/less.js/blob/eefe33a47f6fdcc228817df7435a1770ce9e51ea/packages/less/test/less-test.js#L115-L119

Although I think this actually refers to the wrong error object? I think this should refer to the variable `e` that comes from the `fs.readFile` callback, because `err` (passed as an arg to the function) is handled at the beginning of the function: https://github.com/less/less.js/blob/eefe33a47f6fdcc228817df7435a1770ce9e51ea/packages/less/test/less-test.js#L87-L90

This happens in several of the tests, so I just followed the convention. If it's correct that this should, in fact, be `e` (and `e.message`/`e.stack`), I'm happy to follow up with a PR that fixes these references throughout this file 🙂

Reply:
> Although I think this actually refers to the wrong error object?

@pgoldberg That's entirely possible. I'd like to see all of the [source entirely type-checked with JSDoc annotations](https://github.com/less/less.js/discussions/3786), because then anyone contributing would not have to guess what is what.